core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -678,4 +686,8 @@ private void decoratePillarWithJdbcConnectors(Cluster cluster, Map<String, SaltP
                 .flatMap(Collection::stream)
                 .collect(Collectors.toList());
     }
+
+    private ClouderaManagerRepo getClouderaManagerRepo(final Long clusterId) {
+       return clusterComponentConfigProviderMap.computeIfAbsent(clusterId, k -> clusterComponentConfigProvider.getClouderaManagerRepoDetails(clusterId));
```

Comment:
This is a singleton bean, we can't store any data in it. This call can be moved to `addClouderaManagerConfig` and assign it to a method local variable and pass it as a parameter to whichever method needs it.

Reply:
i see, looks like the call needs to be mover further up to `createSaltConfig` method.